src/Storages/MergeTree/MergeTreeReaderWide.cpp

```diff
@@ -114,7 +116,9 @@ size_t MergeTreeReaderWide::readRows(size_t from_mark, bool continue_reading, si
                 size_t column_size_before_reading = column->size();
                 auto & cache = caches[column_from_part.getNameInStorage()];
 
-                readData(column_from_part, column, from_mark, continue_reading, max_rows_to_read, cache);
+                readData(
+                    column_from_part, column, from_mark, continue_reading,
+                    max_rows_to_read, cache, /* was_prefetched =*/ !prefetched_streams.empty());
```

Comment:
`prefetched_streams` can be empty here only if there was an exception for all `num_columns` in line 90? Why is it possible?

Reply:
Discussed in TG